src/tools/create/code_from_description.ts

```diff
@@ -0,0 +1,106 @@
+import * as vscode from 'vscode';
+import { OpenAIRepository } from '../../repository/openai-repository';
+import { extractDartCode } from '../../utilities/code-processing';
+
+export async function createCodeFromDescription(openAIRepo: OpenAIRepository) {
+    try {
+        const editor = vscode.window.activeTextEditor;
+        if (!editor) {
+            vscode.window.showErrorMessage('No active editor');
+            return;
+        }
+
+        var { aboveText, belowText, cursorLine } = getCodeAroundCursor(editor);
+
+        const instructions = await vscode.window.showInputBox({ prompt: "Enter refactor instructions" });
+        if (!instructions) {
+            return;
+        }
+
+        await vscode.window.withProgress({
+            location: vscode.ProgressLocation.Notification,
+            title: "Generating code",
+            cancellable: false
+        }, async (progress) => {
+            let progressPercentage = 0;
+            let prevProgressPercentage = 0;
+            const progressInterval = setInterval(() => {
+                prevProgressPercentage = progressPercentage;
+                progressPercentage = (progressPercentage + 10) % 100;
+                const increment = progressPercentage - prevProgressPercentage;
+                progress.report({ increment });
+            }, 200);
+            let prompt = `You're an expert Flutter developer. Follow the instructions carefully and to the letter.\n\n`;
```

Comment:
```suggestion
            let prompt = `You're an expert Flutter/Dart coding assistant. Follow the user instructions carefully and to the letter.\n\n`;
```
Keeping the identity consistent across all platforms.